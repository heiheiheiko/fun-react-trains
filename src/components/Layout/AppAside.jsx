import React from 'react';
import TrainFilterForm from './TrainFilterForm';
import MapZoomControls from './MapZoomControls';
import MapCenterLabel from './MapCenterLabel';

function AppAside(props) {
  return (
    <div className="overlay__layer overlay__layer--top-right mt-small">
      <div className="py-small pr-large pl-normal has-background-light">
        <TrainFilterForm/>
      </div>

      <div className="mt-small py-small pr-large pl-normal has-background-light">
        <MapZoomControls/>
      </div>

      <div className="mt-small py-small pr-large pl-normal has-background-light">
        <MapCenterLabel/>
      </div>
    </div>
   );
}

export default AppAside;
