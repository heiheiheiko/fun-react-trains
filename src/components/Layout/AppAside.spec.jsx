import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from "react-redux";
import renderer from 'react-test-renderer'

import AppAside from './AppAside';

jest.mock('./TrainFilterForm', () => ()=> <div> TrainFilterForm </div>);
jest.mock('./MapZoomControls', () => ()=> <div> MapZoomControls </div>);
jest.mock('./MapCenterLabel', () => ()=> <div> MapCenterLabel </div>);

describe('<AppAside>', () => {
  const initialState = { mapCenter: [1,1] };
  const mockStore = configureStore();
  let store, componentContainer;

  beforeEach(() => {
    store = mockStore(initialState)
    componentContainer = <Provider store={store}><AppAside /></Provider>;
  })

  it('could be renderd', () => {
    expect(mount(componentContainer).find(AppAside).length).toEqual(1)
  });

  it('matched the snapshot', () => {
    const renderedValue =  renderer.create(componentContainer).toJSON()
    expect(renderedValue).toMatchSnapshot();
  });
});