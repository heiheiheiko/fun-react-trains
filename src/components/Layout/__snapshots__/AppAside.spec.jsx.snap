// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppAside> matched the snapshot 1`] = `
<div
  className="overlay__layer overlay__layer--top-right mt-small"
>
  <div
    className="py-small pr-large pl-normal has-background-light"
  >
    <div>
       TrainFilterForm 
    </div>
  </div>
  <div
    className="mt-small py-small pr-large pl-normal has-background-light"
  >
    <div>
       MapZoomControls 
    </div>
  </div>
  <div
    className="mt-small py-small pr-large pl-normal has-background-light"
  >
    <div>
       MapCenterLabel 
    </div>
  </div>
</div>
`;
